main.go

```diff
@@ -291,66 +290,49 @@ func (b *Backend) updateCallStats(t shortTraceMsg) {
 
 type loadBalancer struct {
 	backends  []*Backend
-	next      int // next backend the request should go to.
 	listIndex int // List happens on the same backend always as S3 list is stateful for MinIO.
-	sync.RWMutex
 }
 
 // Returns the next backend the request should go to.
-func (lb *loadBalancer) nextProxy() (*httputil.ReverseProxy, int) {
-	lb.Lock()
-	defer lb.Unlock()
-
+func (lb *loadBalancer) nextProxy() *Backend {
 	tries := 0
-	idx := 0
 	for {
-		var proxy *httputil.ReverseProxy
-		if lb.backends[lb.next].up {
-			proxy = lb.backends[lb.next].proxy
-			idx = lb.next
-		}
-		lb.next++
-		if lb.next == len(lb.backends) {
-			lb.next = 0
-		}
-		if proxy != nil {
-			return proxy, idx
+		idx := rng.Intn(len(lb.backends))
+		if lb.backends[idx].IsUp() {
+			// random backend is up.
+			return lb.backends[idx]
 		}
 		tries++
 		if tries == len(lb.backends) {
+			// no backend is up we fail the request.
```

Comment:
previous round robin LB is better. For ex, out of 10 nodes, let's say first 3 are down, then the 4th server gets 40% of the load

Reply:
Load balancer is not here to optimize down node situation @krishnasrinivas it is meant to evenly spread the requests..

Sidekick's purpose was to do randomized load balancing other than listing calls. Currently sidekick is not doing that, round robin is not optlmal is good situations.

Reply:
In fact, this was the main requirement of sidekick implementation, if you look at the distribution now you can actually see the best distribution possible even with the downed situation too. 

Due to randomness involved, it allows for a much wider spread. round-robin is generally prone to problems of overloading a single server during disconnects.

Reply:
I shall add affinity for the number of tries such that tries are not exhausted due to randomness.

Reply:
I made more changes @krishnasrinivas PTAL